feature_engine/encoding/mean_encoding.py

```diff
@@ -143,8 +156,20 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
 
         self.encoder_dict_ = {}
 
+        y_mean = y.mean()
+        if self.a == 'auto':
+            y_var = y.var(ddof=0)
         for var in self.variables_:
-            self.encoder_dict_[var] = y.groupby(X[var]).mean().to_dict()
+            if self.a == 'auto':
+                damping = y.groupby(X[var]).var(ddof=0) / y_var
+            else:
+                damping = self.a
+            counts = X[var].value_counts()
+            smoothing = counts / (counts + damping)
```

Comment:
In category encoders, the smoothing parameter is weighted by the number of observations per category:
https://github.com/scikit-learn-contrib/category_encoders/blob/80288a0aa3d9bc4406f3d6df13a49113bbc3fedf/category_encoders/target_encoder.py#L229-L231

Reply:
I did not use sigmoid formula, I used (x / (x+smoothing)) formula, which does not have this parameter

Reply:
Oh, I see, so `damping = y.groupby(X[var]).var(ddof=0) / y_var` is equation 6 in the article and `x / (x+smoothing)` is equation 5, am I right? Both in section 3.1

Any particular reason you chose this implementation?

I think it might be useful to offer some guidance regarding what are good values of a, if they exist. Or is it just by cross-validation and hyperparamter search?

Reply:
I choose this formula for two reasons:
1. Almost any implementation of target encoding uses sigmoid formula (category_encoders, dirty_cat etc), and it's meaningless to just reimplement it for 100th time.
2. This formula has an interesting auto option that I've never encountered in target encoding estimators (or maybe I searched not good enough, idk), so it seemed interesting.

As for meaningful smoothing values, I think in general it's goes like this "the more noise you have in target, the bigger value of smoothing should be", but there is no advice on concrete values. Most of the time its optimized by cv, sometimes its just set as 1. We will have an auto option for that.

Reply:
Makes sense. Thank you!